server/controllers/bucketlist.controller.js

```diff
@@ -1,7 +1,22 @@
 const BucketList = require('../models/bucketlist.model');
 const User = require('../models/usermodel');
 
-//Create a new Bucketlist
+//Function for checking if a bucketlist is associated with a user
+function confirmDataAssociation(bucketlistId, userBucketlists){
+    let result = false;
+
+    //Loop through all the bucketlists belonging to the user
+    for(var i = 0; i < userBucketlists.length; i++){
```

Comment:
Why do we need this method? I am thinking about performance here. if a user has 1000 bucket list items are we going to loop through 1000 items?

Reply:
I will enhance the performance of the application by ending the loop once we find the bucketlist. This means that we do not have to keep on looping even when our bucketlist has already been found.

Reply:
I have incorporated your valuable feedback and reduced the performance problem.